### ISSUE TITLE
Reliance   on java.util.random

### ISSUE THREAD

**Issue by dvberkel** [phedny/Mammon#9]
The [code](https://github.com/phedny/Mammon/blob/master/interfaces/src/main/java/org/mammon/brands/Group.java#L29) relies on java.util.random.

This is untestable so I would like to propose an interface `RandomGenerator` and wrapper implementations.
Reliance

**Comment by phedny** [comment]
Two remarks I want to make:
- I relied on java.util.Random, such that the user of the interface can decide whether he wants to put in a java.util.Random or java.security.SecureRandom object;
- Another method that is missing, which I thought about after I committed the earlier version, is a method like Group.deserializeElement() which accepts a (JSON?) serialized representation of the element. Likewise, a Group.Element.serializeElement() would be the reverse operation. These are at least going to be used when storing information into persistant storage or transferring elements over a network connection.

During testing, you might find it sufficient (although I doubt it) to use a java.util.Random object initialized with a known seed, such that the values it produces are repeatable. Otherwise, you could use the deserialization-method to instantiate group elements with a known value, which should therefore be very useful for testing.

Please make the necessary modifications if you decide it's better.

**Comment by dvberkel** [comment]
You are correct in your assumption that I don't think seeding the Random number generator is sufficient. I will make the modifications.

As to serializing and deserializing, Lets wait till we really need to do this. Mayby we can get away without it.

**Comment by phedny** [comment]
I don't think we can get around (de)serializing, since group elements must be transferred between parties. For example, the public key of a bank is a group element, the blinded identity of an account holder is a group element, unspent coins consist of two group members and a signature, which in turn consists of four group members.

**Comment by dvberkel** [comment]
I see now that we can't do without (de)serializing. But we could still discuss the method.

I like the [renderContentOn](http://www.seaside.st/about/examples/counter) message of [Seaside](http://www.seaside.st/), a Smalltalk web application framework.

Would you agree if we use a similar construction for (de)serialisation?

**Comment by dvberkel** [comment]
I think we can close this issue. My test zealousness has wained a bit with respect to java.util.random
